app/controllers/admin/lessons_controller.rb

```diff
@@ -0,0 +1,52 @@
+class Admin::LessonsController < ApplicationController
+  before_action :require_admin
+  before_action :find_lesson, only: [:destroy, :show]
+
+  def index
+    if params[:lesson]
```

Comment:
```suggestion
   @lesson = if params[:lesson]
       Lesson.search(params[:lesson])
       else
       Lesson.all
       end
```

nhưng thường c k hay viết như này, c sẽ viết trong thằng search kia, nếu truyền lên rỗng => search all